kernel/pte_flags/src/pte_flags_aarch64.rs

```diff
@@ -0,0 +1,168 @@
+        const DEVICE_MEMORY      = Self::MAIR_INDEX_0.bits;
+        /// Indicates the page's cacheability is described by MAIR Index 1.
+        /// Theseus uses this index for "device" memory.
+        const MAIR_INDEX_1       = 1 << 2;
+        /// This page maps "normal" memory, i.e., non-device memory.
+        /// Theseus uses `MAIR_INDEX_1` for this type of memory.
+        const NORMAL_MEMORY      = Self::MAIR_INDEX_1.bits;
+        /// Indicates the page's cacheability is described by MAIR Index 2.
+        /// This is unused in Theseus.
+        const _MAIR_INDEX_2      = 2 << 2;
+        /// Indicates the page's cacheability is described by MAIR Index 3.
+        /// This is unused in Theseus.
+        const _MAIR_INDEX_3      = 3 << 2;
+        /// Indicates the page's cacheability is described by MAIR Index 4.
+        /// This is unused in Theseus.
+        const _MAIR_INDEX_4      = 4 << 2;
+        /// Indicates the page's cacheability is described by MAIR Index 5.
+        /// This is unused in Theseus.
+        const _MAIR_INDEX_5      = 5 << 2;
+        /// Indicates the page's cacheability is described by MAIR Index 6.
+        /// This is unused in Theseus.
+        const _MAIR_INDEX_6      = 6 << 2;
+        /// Indicates the page's cacheability is described by MAIR Index 7.
+        /// This is unused in Theseus.
+        const _MAIR_INDEX_7      = 7 << 2;
+
+        /// * If set, this page is accessible in both Secure and Non-Secure execution levels.
+        /// * If not set, this page is accessible in only Secure execution levels.
+        /// 
+        /// This is unused in Theseus.
+        const _NON_SECURE_ACCESS = 1 << 5;
+
+        /// * If set, userspace (unprivileged mode) can access this page.
+        /// * If not set, only kernelspace (privileged mode) can access this page.
+        const _USER_ACCESSIBLE   = 1 << 6;
+
+        /// * If set, this page is read-only.
+        /// * If not set, this page is writable.
+        const READ_ONLY          = 1 << 7;
+
+        /// Indicates that only a single CPU core may access this page.
+        const _NON_SHAREABLE     = 0 << 8;
+        // Shareable `0b01` is reserved.
+        // const SHAREABLE_RSVD  = 1 << 8;
+        /// Indicates that multiple CPUs from multiple clusters may access this page.
+        /// This is the default and the the only value used in Theseus (and most systems).
+        const OUTER_SHAREABLE    = 2 << 8;
+        /// Multiple cores from the same
+        /// cluster can access this page.
+        /// Indicates that multiple CPUs from only a single cluster may access this page.
+        const _INNER_SHAREABLE   = 3 << 8;
+
+        /// * The hardware will set this bit when the page is accessed.
+        /// * The OS can then clear this bit once it has acknowledged that the page was accessed,
+        ///   if it cares at all about this information.
+        const ACCESSED           = 1 << 10;
```

Comment:
I think this is incorrect: without setting the bit, qemu's emulated CPU refused to insert the descriptor in the TLB, and this resulted in page faults. By [manually setting the bit](https://github.com/NathanRoyer/aarch64-uefi/blob/main/uefi/src/memory.rs#L79), the descriptor was accepted into the TLB and I encountered no page fault. Maybe this was because of my configuration of translation control registers though? I'll check. It could also be qemu not implementing FEAT_HAFDBS.

Reply:
should be fixed now